client/src/components/AvgRating.jsx

```diff
@@ -3,29 +3,43 @@ import StarRating from './StarRating.jsx';
 import Atelier from '../Atelier.js';
 
 const AvgRating = ({ productId }) => {
-  const [ratings, setRatings] = useState([]);
+  const [ratings, setRatings] = useState({});
   const [avgScore, setAverage] = useState(0);
 
-  const averageScore = (array) => {
-    const average = Math.round(array.reduce((a, b) => Number(a) + Number(b) / array.length)).toFixed(2);
-    return average;
+  const averageScore = (data) => {
+    let sum = 0;
+    let ratings = 0;
+    for (let number in data) {
+      sum += Number(number) * Number(data[number]);
+      console.log(sum);
+      ratings += Number(data[number]);
+      console.log(ratings);
+    }
+    return (sum / ratings).toFixed(2);
   };
 
   const fetchRatings = async () => {
```

Comment:
Shouldn't this be getting the metadata from App via props?

Reply:
it does not b/c this is used on each of the product cards, so those products have different ratings